### PR TITLE
nixos/*: use pipewire by default

### PR DESCRIPTION
**Pull request by K900**
It's generally more reliable than Pulse these days, and Pulse also has a weird interaction with Plasma where the sound is muted on boot.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix [hunk 1]
```diff
@@ -33,7 +33,11 @@ with lib;
   powerManagement.enable = true;
 
   # Enable sound in graphical iso's.
-  hardware.pulseaudio.enable = true;
+  services.pipewire = {
+    enable = true;
+    pulse.enable = true;
+    alsa.enable = true;
```
Comment (SuperSandro2000) [inline nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix:39]: alsa is free to enable, right?

Comment (K900) [reply nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix:39]: Almost. It's a few kilobytes at most.

Comment (Mic92) [reply nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix:39]: Should we than also enable this by default in gnome.nix as a follow up?

Comment (Mic92) [reply nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix:39]: Also don't we need to disable pulseaudio here because gnome enables it?

Comment (K900) [reply nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix:39]: That's a good question. The Plasma modules default to "no opinion" currently, but maybe we should switch everything to Pipewire?

Comment (Aleksanaa) [reply nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix:39]: People should expect the live ISO to have the same default configuration as the installed system.

Comment (Mic92) [reply nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix:39]: I am ok with having this in GNOME and KDE enabled as well.

**Comment by K900** [comment]
Go hard or go home. Pipewire defaults everywhere. May the bikes be shed.

**Review by Mic92 — APPROVED** [review]

**Review by jopejoe1 — APPROVED** [review]

**Review by emilazy — APPROVED** [review]
🚀

**Review by Qubasa — APPROVED** [review]

**Review by mkg20001 — APPROVED** [review]

**Comment by rnhmjoj** [comment]
This completely breaks ALSA and is very hard to find out why.
I guess the problem is not really this change per-se, but the fact that just enabling the xserver this `graphical-desktop.nix` module gets pulled (enabling all sort of stuff). It should really be enabled only by selected desktop environments.

**Comment by SuperSandro2000** [comment]
But pipewire is enabling asla, too. Ist that not working for you?

**Comment by rnhmjoj** [comment]
> But pipewire is enabling asla, too. Ist that not working for you?

No, for some reason pipewire is not being started and attempts to produce audio results in errors like "Host is down".

**Pull request merged**